clients/fides-js/src/lib/automated-consent.ts

```diff
@@ -84,97 +68,64 @@ export const automaticallyApplyPreferences = async ({
     return false;
   }
 
-  /**
-   * Select the "best" translation that should be used for these saved
-   * preferences based on the currently active locale.
-   *
-   * NOTE: This *feels* a bit weird, and would feel cleaner if this was moved
-   * into the UI components. However, we currently want to keep the GPC
-   * application isolated, so we need to duplicate some of that "best
-   * translation" logic here.
-   */
-  const bestTranslation = selectBestExperienceConfigTranslation(
-    i18n,
-    effectiveExperience.experience_config,
-  );
-  const privacyExperienceConfigHistoryId =
-    bestTranslation?.privacy_experience_config_history_id;
-
   let gpcApplied = false;
   let noticeConsentApplied = false;
   let migratedConsentApplied = false;
 
-  const consentPreferencesToSave = effectiveExperience.privacy_notices.map(
-    (notice) => {
+  const noticeConsentToSave = experience.privacy_notices.reduce(
+    (accumulator, notice) => {
+      const appliedConsent = { ...accumulator };
+      const defaultBoolean = transformUserPreferenceToBoolean(
+        notice.default_preference,
+      );
+      appliedConsent[notice.notice_key] = defaultBoolean;
+      if (savedConsent[notice.notice_key]) {
+        appliedConsent[notice.notice_key] = savedConsent[notice.notice_key];
+      }
       const hasPriorConsent = noticeHasConsentInCookie(notice, savedConsent);
-      const bestNoticeTranslation = selectBestNoticeTranslation(i18n, notice);
+      const isNoticeOnly =
+        notice.consent_mechanism === ConsentMechanism.NOTICE_ONLY;
 
       // First check for migrated consent
       if (hasMigratedConsent && migratedConsent) {
         const preference = migratedConsent[notice.notice_key];
         if (preference !== undefined) {
           migratedConsentApplied = true;
-          const userPreference =
-            typeof preference === "boolean"
```

Comment:
Wow this file is so much cleaner now! I really like that we let `preferences.ts` handle `transformConsentToFidesUserPreference`, so that much of this duplicate code can be removed from this file 🌟